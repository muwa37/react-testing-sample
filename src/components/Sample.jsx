import React, { useEffect, useState } from 'react';

const Sample = () => {
  const [data, setData] = useState(null);
  const [toggle, setToggle] = useState(false);
  const [value, setValue] = useState('');

  const onToggle = () => setToggle(prev => !prev);

  useEffect(() => {
    setTimeout(() => {
      setData({});
    }, 100);
  }, []);

  return (
    <div>
      {toggle && <div data-testid='toggle-elem'>toggle</div>}
      <h2>hello world</h2>
      <button data-testid='toggle-btn' onClick={onToggle}>
        click on me to toggle
      </button>
      <h3 data-testid={'value-elem'}>{value}</h3>
      <input
        type='text'
        placeholder='type in me...'
        value={value}
        onChange={e => setValue(e.target.value)}
      />
      {data && <div style={{ color: 'red' }}>data</div>}
    </div>
  );
};

export default Sample;
