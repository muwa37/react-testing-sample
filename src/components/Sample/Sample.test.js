import { fireEvent, render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import Sample from './Sample';

describe('test sample component', () => {
  test('sample', () => {
    render(<Sample />);

    const helloWorldEl = screen.getByText(/hello world/i);
    const btnEl = screen.getByRole('button');
    const inputEl = screen.getByPlaceholderText(/type in me.../i);

    expect(helloWorldEl).toBeInTheDocument();
    expect(btnEl).toBeInTheDocument();
    expect(inputEl).toMatchSnapshot();
  });

  test('sample async and styles', async () => {
    render(<Sample />);
    const asyncDataEl = await screen.findByText(/data/i);
    expect(asyncDataEl).toBeInTheDocument();
    expect(asyncDataEl).toHaveStyle({ color: 'red' });
  });

  test('sample testid with event', () => {
    render(<Sample />);
    const btnByTestId = screen.getByTestId('toggle-btn');
    expect(screen.queryByTestId('toggle-elem')).toBeNull();
    fireEvent.click(btnByTestId);
    expect(screen.getByTestId('toggle-elem')).toBeInTheDocument();
    fireEvent.click(btnByTestId);
    expect(screen.queryByTestId('toggle-elem')).toBeNull();
  });

  test('sample input event', () => {
    render(<Sample />);
    const inputEl = screen.getByPlaceholderText(/type in me.../i);
    expect(screen.queryByTestId('value-elem')).toContainHTML('');
    userEvent.type(inputEl, 'test');
    expect(screen.queryByTestId('value-elem')).toContainHTML('test');
  });
});
