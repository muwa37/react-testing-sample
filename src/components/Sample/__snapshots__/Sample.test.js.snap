// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test sample component sample 1`] = `
<input
  placeholder="type in me..."
  type="text"
  value=""
/>
`;
